pydicom/dataset.py

```diff
@@ -179,10 +179,15 @@ def add_new(self, element_offset, VR, value):
         self.dataset.add_new(self.get_tag(element_offset), VR, value)
 
 
-class Dataset(object):
+class Dataset(dict):
     """Contains a collection (dictionary) of DICOM DataElements.
     Behaves like a dictionary.
 
+    .. note::
+
+        `Dataset` is derived from `dict` only to make it work in a NumPy
+        array. No `dict` functionality is used.
+
```

Comment:
Well... dict functionality is very much used through the wrapped dict instance.  Perhaps wording could be "the parent dict class is never called, as all dict methods are overridden"

Reply:
..although are we sure of that?  That they are _all_ overridden?  If not, some very strange behavior would occur - a populated dataset acting as if empty.

Reply:
I tried this with this branch:
```python
>>> from pydicom.dataset import Dataset
>>> ds = Dataset()
>>> di = dict()
>>>
>>> set(dir(di)) - set(dir(ds))
{'__class__', '__doc__', '__hash__'}

```
The `__class__`, and `__doc__` gives Dataset's expected results, and hash returns None as required.  So it seems there is nothing in the parent class that can be called...  But now I wonder about the super() calls that still exist in dataset.py - is that still the correct thing to do?   If `dict` gets some extra methods in future python versions, the super() calls could pass them through.

Reply:
Good point! I will change these. Also, it makes sense to add your test for `dict` methods that are not overwritten to make sure it doesn't change in future versions.

Reply:
Ok, done.